File: services/typescript/express/src/routes/handlers/invalid_method.ts
import type { RequestHandler } from "express";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "telemetry";

export const invalidMethodHandler: RequestHandler = (req, res) => {
  const logger = getLogger();
  logger.warn({ method: req.method, path: req.path }, "Invalid method");
  sendError(res, AppErrors.InvalidMethod);
};

File: services/python/django/src/routes/handlers/invalid_method.py
import structlog
from django.http import HttpRequest, HttpResponse

from routes.errors import AppErrors, send_error


def invalid_method_handler(request: HttpRequest, _error=None) -> HttpResponse:
    structlog.get_logger().warn(
        "Invalid method", method=request.method, path=request.path
    )
    return send_error(AppErrors.InvalidMethod)
